myfitnesspal/client.py

```diff
@@ -804,3 +810,535 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+
+        # Get User Default Unit Preference
+        if energy_unit != "calories" and energy_unit != "kilojoules":
+            energy_unit = self.user_metadata["unit_preferences"]["energy"]
+
+        # Get authenticity token and current values
+        url = parse.urljoin(self.BASE_URL_SECURE, "account/my_goals")
+        today = datetime.datetime.now().strftime("%Y-%m-%d")
+
+        """
+        Will leave this here as reminder why we do not use self._get_json_for_url() at the moment.
+        # No Way implemented to send token by calling self._get_json_for_url
+        old_goals_url = parse.urljoin(self.BASE_API_URL, f"v2/nutrient-goals?date={today}")
+        old_goals = self._get_json_for_url(old_goals_url)
+        """
+
+        # Build header for API-requests
+        auth_header = self.session.headers
+        auth_header["authorization"] = f"Bearer {self.access_token}"
+        auth_header["mfp-client-id"] = "mfp-main-js"
+        auth_header["mfp-user-id"] = f"{self.user_id}"
+
+        # Get Request for old goal values
+        old_goals_url = parse.urljoin(
+            self.BASE_API_URL, f"v2/nutrient-goals?date={today}"
+        )
+        old_goals_document = self.session.get(old_goals_url, headers=auth_header)
+        old_goals = json.loads(old_goals_document.text)
+
+        # Marcro Calculation
+        # If no macro goals were provided calculate them with percentage value
+        if carbohydrates is None or protein is None or fat is None:
+            # If even no macro percentages values were provided calculate them from old values
+            if (
+                percent_carbohydrates is None
+                or percent_protein is None
+                or percent_fat is None
+            ):
+                if energy is None:
+                    logger.warning("No energy value and no macro values provided! - Not able to update goals without a mandotory amount of information.")
+                    return False
+                else:
+                    old_energy_value = old_goals["items"][0]["default_goal"]["energy"][
+                        "value"
+                    ]
+                    old_energy_unit = old_goals["items"][0]["default_goal"]["energy"][
+                        "unit"
+                    ]
+                    old_carbohydrates = old_goals["items"][0]["default_goal"][
+                        "carbohydrates"
+                    ]
+                    old_fat = old_goals["items"][0]["default_goal"]["fat"]
+                    old_protein = old_goals["items"][0]["default_goal"]["protein"]
+
+                    # If old and new values are in diffrent units then convert old value to new unit
+                    if not old_energy_unit == energy_unit:
+                        if old_energy_unit == "kilojoules" and energy_unit == "calories":
+                            old_energy_value *= 0.2388
```

Comment:
I'm sure it's not going to matter much, but I _believe_ the actual value is 0.239006, no?

Reply:
I took the values from the MFP javascript to match the original behaviour. :)